docs/en/observability/apm/configure/outputs/logstash.asciidoc

```diff
@@ -129,23 +129,40 @@ In addition to `@metadata`, APM Server provides other potentially useful fields,
 `data_stream` field, which can be used to conditionally operate on
 {observability-guide}/apm-data-model.html[event types], namespaces, or datasets.
 
-As an example, you might want to use {ls} to route all `metric` events to the same custom metrics data stream,
-rather than to service-specific data streams:
+As an example, you might want to use {ls} to route all `metrics` events to the same custom metrics data stream,
+rather than to service-specific data streams.
+
+However, if when you combine all `metrics` events there are events that have the `data_stream.dataset` field
+set to different values, indexing will fail with a message stating that the field does not accept any other values.
+This is because the `data_stream.dataset` field's mapping is set to `constant_keyword`,
+which expects all values of the fields in the index to be the same.
```

Comment:
Is this accurate?

Reply:
Yes, this is accurate.

> Is there a specific error/warning message we can include in case any users are searching for information about it?

For example, try running the following two requests:

```
POST /metrics-custom-default/_doc
{
  "@timestamp": "2024-10-08",
  "data_stream": {"dataset": "custom"}
}

POST /metrics-custom-default/_doc
{
  "@timestamp": "2024-10-08",
  "data_stream": {"dataset": "standard"}
}
```

The first one will auto-create the data stream, and the initial value of `data_stream.dataset: custom` will be used for the `constant_keyword` value. Any future values must match this. The second one does not match, and will fail with an error like:

```
{
  "error": {
    "root_cause": [
      {
        "type": "document_parsing_exception",
        "reason": "[1:60] failed to parse field [data_stream.dataset] of type [constant_keyword] in document with id '2z29apIBiisszFNu21Dk'. Preview of field's value: 'standard'"
      }
    ],
    "type": "document_parsing_exception",
    "reason": "[1:60] failed to parse field [data_stream.dataset] of type [constant_keyword] in document with id '2z29apIBiisszFNu21Dk'. Preview of field's value: 'standard'",
    "caused_by": {
      "type": "illegal_argument_exception",
      "reason": "[constant_keyword] field [data_stream.dataset] only accepts values that are equal to the value defined in the mappings [custom], but got [standard]"
    }
  },
  "status": 400
}
```

So the error messages to look for would be:
 - `failed to parse field [data_stream.dataset] of type [constant_keyword]`
 - `[constant_keyword] field [data_stream.dataset] only accepts values that are equal to the value defined in the mappings`